src/Context/snackbarProvider.tsx

```diff
@@ -0,0 +1,45 @@
+import { createContext, useContext, useReducer, FC } from "react";
+import { Snackbar, Alert, AlertColor } from "@mui/material";
+import { snackbarReducer } from "../Reducers/snackbarReducer";
+import { initialSnackbar } from "../constants/initialSnackbar";
+import { SnackbarContextType, SnackbarProviderProps } from "../types/snackbar";
+
+const SnackbarContext = createContext<SnackbarContextType | undefined>(
+  undefined
+);
+
+export const SnackbarProvider: FC<SnackbarProviderProps> = ({ children }) => {
+  const [state, dispatch] = useReducer(snackbarReducer, initialSnackbar);
+
+  const showSnackbar = (message: string, severity: AlertColor = "info") => {
```

Comment:
You dont need to handle the state here and inside the reducer we can do the check on the root ( the reducer ) and that would be enough. 

```suggestion
  const showSnackbar = (message: string, severity: AlertColor) => {
```